### DISCUSSION TITLE
Do tree files need to follow strict hierarchy?

### DISCUSSION THREAD

**Discussion by PraneshASP** [alexfertel/bulloak#55] [Q&A] [answered]
Is there an hierarchy for the tree files? `When -> Given -> It` or `when` and `given` can be in the same level as well? I was working on adding custom folding for tree files based to my extension but not sure if there is an hierarchy for the tree. 

For ex., in the tree below, `when` and `given` are in the same level. 

```tree
├── when `aToken._mintScaled` results in a scaled amount that's zero (prank test)
│  └── it reverts
├── given incentivesController != address(0) in `MintableIncentivizedERC20._mint`
│  └── it ...s TODO: Handle this case
└── given incentivesController == address(0)
   ├── given this is the first supply
```

So I guess it should be as below?

```tree
├── when `aToken._mintScaled` results in a scaled amount that's zero (prank test)
│  └── it reverts
├── when incentivesController != address(0) in `MintableIncentivizedERC20._mint`
│  └── it ...s TODO: Handle this case
└── when incentivesController == address(0)
   ├── given this is the first supply
```

**Comment by PaulRBerg** [comment, marked as answer]
There are no hierarchy rules. See my [BTT examples](https://github.com/PaulRBerg/btt-examples) repo.

`when` can come before `given`, and `given` can come before `when`.

**Reply by PraneshASP** [reply]
Gotcha! Thanks for confirming.
